module/icp/io/sha2_mod.c

```diff
@@ -591,7 +591,7 @@ sha2_digest_atomic(crypto_provider_handle_t provider,
 {
 	int ret = CRYPTO_SUCCESS;
 	SHA2_CTX sha2_ctx;
-	uint32_t sha_digest_len;
+	uint32_t sha_digest_len = SHA2_HMAC_MIN_KEY_LEN;
```

Comment:
I don't see how this could be uninitialized when it's used.  It must either be SHA512_DIGEST_LENGTH or SHA256_DIGEST_LENGTH based on the if/else conditional on line 666.